crml/staking/src/rewards/mod.rs

```diff
@@ -151,17 +172,88 @@ where
 				Self::calculate_npos_payouts(&validator, *validator_commission, stake_map, total_payout_share)
 			})
 			.for_each(|(account, payout)| {
-				total_payout_imbalance.maybe_subsume(T::CurrencyToReward::deposit_into_existing(&account, payout).ok());
+				Payouts::<T>::mutate(|p| p.push_back((account, payout, era)));
 			});
 
-		// Any unallocated reward amount can go to the development fund
-		let remainder = total_payout.saturating_sub(total_payout_imbalance.peek());
-		total_payout_imbalance.subsume(
-			T::CurrencyToReward::deposit_into_existing(&T::TreasuryModuleId::get().into_account(), remainder)
-				.unwrap_or_else(|_| PositiveImbalanceOf::<T>::zero()),
-		);
+		RemainingRewardAmounts::<T>::mutate(|rra| {
+			rra.push_back(total_payout.saturating_sub(total_payout_imbalance.peek()))
+		});
+	}
+
+	/// Process the reward payouts considering the given quota which is the number of payouts to be processed now.
+	/// Return the benchmarked weight of the call.
+	fn process_reward_payouts(remaining_blocks: Self::BlockNumber) -> Weight {
+		let remaining_payouts = Payouts::<T>::get().len() as u32;
+		let quota = Self::calculate_payout_quota(remaining_payouts, remaining_blocks);
+		if quota.is_zero() {
+			return T::WeightInfo::process_zero_payouts();
+		}
+
+		let weight = T::WeightInfo::process_reward_payouts(quota as u32);
+
+		Payouts::<T>::mutate(|p| {
+			// First payout in the current series. Without separating the first payout,
+			// I couldn't set total_payout_imbalance correctly. It's because simply assigning
+			// PositiveImbalanceOf::<T>::zero() to it, would not set the underlying asset id correctly.
+			let (first_payee, first_amount, first_era) = p.pop_front().unwrap_or_default();
+			let mut total_payout_imbalance = T::CurrencyToReward::deposit_into_existing(&first_payee, first_amount)
+				.ok()
+				.unwrap_or_else(PositiveImbalanceOf::<T>::zero);
+			Self::deposit_event(RawEvent::RewardPayout(first_payee, first_amount, first_era));
+			let mut era_under_process = first_era;
+			let mut era_reward_amount = first_amount;
+
+			let clear_up = |imbalance: &mut PositiveImbalanceOf<T>| -> BalanceOf<T> {
```

Comment:
would be nice if we could queue the treasury payout like any other payout

Reply:
- Will change to the name to `handle_remainder`
- If I want to include deposit event in this lambda then I need to pass era_under_process and era_reward_amount as arguments too. Simply using them inside the lambda distresses the borrow checker. So, in that case, it wouldn't be even neat or useful.
- I used closure because I didn't want to add a function to either the module or the interface, while this function is only used locally here

Reply:
> would be nice if we could queue the treasury payout like any other payout

Treasury payout only happens once in an era and there is no reason to hold it and process it later. So perhaps it's not worth it.